pkg/ccl/backupccl/restore_job.go

```diff
@@ -1146,6 +1150,19 @@ func (r *restoreResumer) Resume(ctx context.Context, execCtx interface{}) error
 	if err != nil {
 		return err
 	}
+	// backupCodec is the codec that was used to encode the keys in the backup. It
+	// is the tenant in which the backup was taken.
+	backupCodec := keys.SystemSQLCodec
+	if len(sqlDescs) != 0 {
+		if len(latestBackupManifest.Spans) != 0 {
+			_, tenantID, err := keys.DecodeTenantPrefix(latestBackupManifest.Spans[0].Key)
```

Comment:
I wonder if we should confirm that all spans, or at least the last span, has the same tenant ID? just thinking to be defensive about backups that contain spans from multiple tenants, where it is unclear that the first is the right one?

Reply:
Good point. Although I think that a backup that has no tenant targets would ensure that the entire keyspace covered by manifest.Spans belongs to 1 tenant?